src/pbkdf2.rs

```diff
@@ -304,4 +304,15 @@ mod tests {
             Ok(())
         });
     }
+
+    #[test]
+    #[should_panic]
+    pub fn pkbdf2_zero_iterations() {
+        let prf = &pbkdf2::HMAC_SHA256;
+        let secret = "ZeroIterationsTest".as_bytes();
+        let iterations = 0;
+        let salt = "salt".as_bytes();
+        let mut out = vec![0u8; 2];
+        pbkdf2::derive(prf, iterations as u32, &salt, &secret, &mut out);
```

Comment:
Thanks!

NIT: don't use `as u32` here. If you need a type annotation then use `let iterations: u32 = 0;` instead.

Also, please make a variant of the test that has all the parameters the same, except iterations >= 1, and which doesn't panic. Add a comment like `// Control for `pkbdf2_zero_iterations()`.

Please add the contributor's statement to the end of your commit message; see https://github.com/briansmith/ring#contributing.

If you know how, please rebase this on master without the ASAN changes, e.g. using `git rebase -i master` and moving this commit to the top of the list. If you don't know how to do that, don't worry about it; I'll do it when I land the change.